frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -124,13 +125,13 @@ export const NavBar = withRouter<NavBarProps>(({pages, basePath, hideDivider}) =
 
   const primaryTabs = <ul className="co-m-horizontal-nav__menu-primary">{
     pages.filter(({href}, i, all) => before.includes(href) || before.includes(_.get(all[i + 1], 'href'))).map(({name, href}) => {
-      const klass = classNames('co-m-horizontal-nav__menu-item', {'co-m-horizontal-nav-item--active': location.pathname.replace(basePath, '/').endsWith(`/${href}`)});
+      const klass = classNames('co-m-horizontal-nav__menu-item', {'co-m-horizontal-nav-item--active':  name === 'Overview' ? location.pathname.replace(basePath, '/').endsWith(`/${href}`) : location.pathname.replace(basePath, '/').includes(`/${href}`)});
```

Comment:
We should use react-router's `matchPath` and `withRouter` `match` param to determine if there is a match instead of doing these string comparisons.
Then you can also make use of the `path` for determining a match.

Reply:
fixed in https://github.com/openshift/console/pull/2695/commits/76d8c8385d320dcb938e94579f1e7a4018d56c70